### ISSUE TITLE
Documentation: Help choosing latex_engine

### ISSUE THREAD

**Issue by JulienPalard** [sphinx-doc/sphinx#4149]
https://github.com/sphinx-doc/sphinx/issues/4136 was "simply" a wrong choice in how to handle UTF-8 characters, leading me to the conclusion: The documentation about `latex_engine` should help users choosing the right engine for their needs.

I'm not a native english speaker, so it may probably be enhanced, but I'd see something like:

```
.. confval:: latex_engine

   The LaTeX engine to build the docs.  The setting can have the following
   values:

   * ``'pdflatex'`` -- PDFLaTeX (default)
   * ``'xelatex'`` -- XeLaTeX (Uses Unicode, good for multilingual support)
   * ``'lualatex'`` -- LuaLaTeX (Uses Unicode, good for multilingual support)
   * ``'platex'`` -- pLaTeX (default if :confval:`language` is ``'ja'``, good at rendering Japanese)
```

**Comment by jfbu** [comment]
Would this be ok?

```diff
diff --git a/doc/config.rst b/doc/config.rst
index e96c075e9..243c374b0 100644
--- a/doc/config.rst
+++ b/doc/config.rst
@@ -1542,6 +1542,17 @@ These options influence LaTeX output. See further :doc:`latex`.
    * ``'lualatex'`` -- LuaLaTeX
    * ``'platex'`` -- pLaTeX (default if :confval:`language` is ``'ja'``)
 
+   Contrarily to ``pdflatex`` which requires fonts prepared for TeX,
+   ``xelatex`` and ``lualatex`` can (and actually, should) use OpenType fonts
+   (from the TeX installation tree or system fonts). Sphinx does not modify
+   the font from its default for these engines: with recent LaTeX this is the
+   Latin Modern font in OpenType format (from the TeX installation tree).
+   These engines are said to be "Unicode" engines as they require UTF-8
+   sources and better support Unicode characters (assuming the glyph is
+   actually provided by the used font). In contrast, ``pdflatex`` support for
+   Unicode characters is often limited to those corresponding to the document
+   language, extra steps being needed to handle others.
+
 .. confval:: latex_documents
 
    This value determines how to group the document tree into LaTeX source files.
```

**Comment by JulienPalard** [comment]
LGTM

**Comment by jfbu** [comment]
Thanks for review; finally I rewrote entirely the added explanations. Fixed at 90e3425ed

**Comment by jfbu** [comment]
Closing.
